### PR TITLE
Avoiding joins

### PR DESCRIPTION
**Pull request by gmazzei**
This PR adds to `SuggestionController` and `UserController`  the same changes done in `BookController` to avoid joins.
This is just an update for the other controllers, so the code in all of them becomes a bit more homogeneous.

### CODE REVIEW

File: Sources/App/Controllers/SuggestionController.swift [hunk 1]
```diff
@@ -23,22 +23,27 @@ final class SuggestionController {
...
+                let suggestionForm = try Suggestion.SuggestionForm(id: suggestion.requireID(),
+                                                                   title: suggestion.title,
+                                                                   author: suggestion.author,
+                                                                   link: suggestion.link,
+                                                                   user: user)
+                promise.succeed(result: suggestionForm)
+            }
+            catch {
+                promise.fail(error: error)
+            }
```
Comment (danielaRiesgo) [inline Sources/App/Controllers/SuggestionController.swift]: Just asking, can't you just chain promises? And ask the `user` of the `futureSuggestion` directly?

File: Sources/App/Controllers/UserController.swift [hunk 2]
```diff
@@ -78,47 +78,67 @@ final class UserController {
...
+        DispatchQueue.global(qos: .default).async {
+            do {
+                let wish = try futureWish.wait()
+                let userCall = wish.user.get(on: req)
+                let bookCall = wish.book.get(on: req)
+                let user = try userCall.wait()
+                let book = try bookCall.wait()
+                
+                let userToCompare = try futureUser.wait()
+                if user.id != userToCompare.id { throw Abort(.notFound) }
```
Comment (danielaRiesgo) [inline Sources/App/Controllers/UserController.swift]: Can;t you compare before anything else so you don't wait for other things you will not do?

Comment (danielaRiesgo) [reply Sources/App/Controllers/UserController.swift]: Oh right. Okay so maybe wait for future User, wait for user, compare and then wait for book

File: Sources/App/Controllers/SuggestionController.swift (hunk 1 above)
Comment (gmazzei) [reply Sources/App/Controllers/SuggestionController.swift]: I think that's the way it was before the changes. I'll leave it as it was before.

**Comment by gmazzei** [comment]
Thanks! :)

**Pull request merged**